x/pool/module.go

```diff
@@ -0,0 +1,140 @@
+package pool
+
+import (
+	"context"
+
+	gwruntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
+
+	modulev1 "cosmossdk.io/api/cosmos/pool/module/v1"
+	"cosmossdk.io/core/appmodule"
+	storetypes "cosmossdk.io/core/store"
+	"cosmossdk.io/depinject"
+	"cosmossdk.io/x/pool/keeper"
+	"cosmossdk.io/x/pool/types"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+// ConsensusVersion defines the current x/consensus module consensus version.
+const ConsensusVersion = 1
+
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
```

Comment:
Let's maybe stop having the app module basic defined and define everything in app module.
Then we can use the core adapter for getting an app module basic if needed. 

https://github.com/cosmosregistry/example/blob/main/module/module.go#L37